src/Mvc/Mvc.Core/src/ModelBinding/Binders/TimeOnlyModelBinder.cs

```diff
@@ -0,0 +1,104 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System.Globalization;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding.Binders;
+
+/// <summary>
+/// An <see cref="IModelBinder"/> for <see cref="TimeOnly"/> and nullable <see cref="TimeOnly"/> models.
+/// </summary>
+public class TimeOnlyModelBinder : IModelBinder
+{
+    private readonly DateTimeStyles _supportedStyles;
+    private readonly ILogger _logger;
+
+    /// <summary>
+    /// Initializes a new instance of <see cref="TimeOnlyModelBinder"/>.
+    /// </summary>
+    /// <param name="supportedStyles">The <see cref="DateTimeStyles"/>.</param>
+    /// <param name="loggerFactory">The <see cref="ILoggerFactory"/>.</param>
+    public TimeOnlyModelBinder(DateTimeStyles supportedStyles, ILoggerFactory loggerFactory)
```

Comment:
Given the duplication between `DateOnly` / `TimeOnly` & `DateTime`, I wonder if it may be appropriate to consider  de-duplicating into one class to simplify future maintenance? Where does everyone land on this?

Reply:
I'm not sure if there's any use case for a customer instantiating an existing model binder, but I'm concerned about introducing a breaking change since the `DateTimeModelBinder` and its corresponding provider are public. What are your thoughts on this?

Reply:
> Given the duplication between `DateOnly` / `TimeOnly` & `DateTime`, I wonder if it may be appropriate to consider de-duplicating into one class to simplify future maintenance? Where does everyone land on this?

I would keep them separate. This is not code that changes frequently and the risk of having a bug in this area is low, which negates a lot of the benefits of having a base class.